lxd/instance/drivers/driver_qemu.go

```diff
@@ -1093,6 +1099,15 @@ func (d *qemu) Start(stateful bool) error {
 		return err
 	}
 
+	// Run any post-start hooks.
```

Comment:
@JLRDRAGON92000 for consistency with lxc containers, it feels to me like the postStartHooks should be run after the database updates below (i.e. UpdateInstancePowerState and UpdateInstanceLastUsedDate) (which normally happen for lxc containers in the OnStart hook).  And after `revert.Success()` so that any reverts setup dont run as they could conflict with the vm.Stop() call on post start hook error.

So I would suggest moving this just above the line `if op.Action() == "start" `, which is exactly where it is for lxc too.

Also, as there are still references to `vm.` here I suspect you need to rebase on master as that has changed to `d.`